packages/mobile/src/tokens/saga.ts

```diff
@@ -73,15 +74,20 @@ export function tokenFetchFactory({ actionName, token, actionCreator, tag }: Tok
       const tokenContract = yield call(getTokenContract, token)
       const balanceInWei: BigNumber = yield call([tokenContract, tokenContract.balanceOf], account)
       const balance: BigNumber = yield call(convertFromContractDecimals, balanceInWei, token)
-      ValoraAnalytics.track(
-        AppEvents.fetch_balance,
+      const balanceLogObject =
         token === CURRENCY_ENUM.DOLLAR
           ? {
               dollarBalance: balance.toString(),
             }
           : { goldBalance: balance.toString() }
-      )
-      yield put(actionCreator(balance.toString()))
+
+      // Only update balances when it's less than the upper bound
+      if (balance.lt(WALLET_BALANCE_UPPER_BOUND)) {
+        yield put(actionCreator(balance.toString()))
+        ValoraAnalytics.track(AppEvents.fetch_balance, balanceLogObject)
+      } else {
+        ValoraAnalytics.track(AppEvents.fetch_balance_error, balanceLogObject)
```

Comment:
I think the `fetch_balance_error` event name is too generic and should be specific about this incorrect high value error.
If someone looks at this from the analytics dashboard it probably won't be clear that it's about this specific case.

Reply:
Do you have a proposal?

I disagree as this follows the naming convention for all other errors we have. We generally do `event_name` and `event_name_error` to capture events related to specific errors. I would rather not break this convention.

Reply:
Assuming this is the only source of errors for fetching a balance it looks fine to me. If we had other sources of errors for fetching balance I'd vote for a field explaining what type of error like `errorType` or something. Not sure if that's necessary here.